aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/BaseHandlerStd.java

```diff
@@ -225,36 +237,47 @@ private ProgressEvent<ResourceModel, CallbackContext> validateModelParameters(fi
+    private ProgressEvent<ResourceModel, CallbackContext> describeCurrentDBClusterParameters(
+            final ProxyClient<RdsClient> proxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress,
+            final Map<String, Parameter> parametersAcc
+    ) {
+        final Map<String, Object> parameters = progress.getResourceModel().getParameters();
+        if (MapUtils.isEmpty(parameters)) {
             return progress;
         }
-        return proxy.initiate("rds::describe-db-parameters", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
-                .translateToServiceRequest(model -> Translator.describeDbClusterParametersFilteredRequest(model, marker))
-                .makeServiceCall((request, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(request,
-                        proxyInvocation.client()::describeDBClusterParameters))
-                .handleError((describeDBParametersRequest, exception, client, resourceModel, ctx) ->
-                        Commons.handleException(
-                                ProgressEvent.progress(resourceModel, ctx),
-                                exception,
-                                DEFAULT_DB_CLUSTER_PARAMETER_GROUP_ERROR_RULE_SET))
-                .done((describeDbClusterParametersRequest, describeDbClusterParametersResponse, proxyInvocation, resourceModel, context) -> {
-                    currentDBClusterParameters.putAll(
-                            describeDbClusterParametersResponse.parameters().stream()
-                                    .collect(Collectors.toMap(Parameter::parameterName, Function.identity(), (v1, v2) -> v2))
-                    );
-
-                    final String nextMarker = describeDbClusterParametersResponse.marker();
-                    if (!StringUtils.isNullOrEmpty(nextMarker)) {
-                        return ProgressEvent.progress(resourceModel, context)
-                                .then(p -> describeCurrentDBClusterParameters(p, currentDBClusterParameters, proxy, proxyClient, nextMarker));
+
+        final ResourceModel model = progress.getResourceModel();
+
+        final List<String> parameterNames = new ArrayList<>(parameters.keySet());
+        for (final List<String> parameterNamePartition : Lists.partition(parameterNames, MAX_PARAMETER_FILTER_SIZE)) {
+            String marker = null;
+            try {
+                do {
+                    final DescribeDbClusterParametersResponse response = fetchDbClusterParameters(proxyClient, model, parameterNamePartition, marker);
+                    for (final Parameter parameter : response.parameters()) {
+                        parametersAcc.put(parameter.parameterName(), parameter);
                     }
-                    return ProgressEvent.progress(resourceModel, context);
-                });
+                    marker = response.marker();
+                } while (marker != null);
```

Comment:
Optional: we can add MAX_DEPTH parameter to avoid infinite loop in case service returned constant marker as flaw?

Reply:
Thanks, will add it